src/ServiceControl.Infrastructure/PluginAssemblyLoadContext.cs

```diff
@@ -0,0 +1,39 @@
+﻿namespace ServiceControl.Infrastructure;
+
+using System.Linq;
+using System.Reflection;
+using System.Runtime.Loader;
+
+public class PluginAssemblyLoadContext(string assemblyPath) : AssemblyLoadContext(assemblyPath)
+{
+    readonly AssemblyDependencyResolver resolver = new(assemblyPath);
+
+    protected override Assembly Load(AssemblyName assemblyName)
+    {
+        if (Default.Assemblies.Any(a => a.FullName == assemblyName.FullName))
```

Comment:
I wasn't sure if this is really something we should focus on. Still mentioning for completness. Do you have an idea how expensive that any is vs have a more efficient data structure like a hashset?

Reply:
I'm not sure how I could have another data structure there since `Default.Assemblies` could contain more items each time `Load` is called. It doesn't seem to make much sense to me to populate a new HashSet each call.

I could swap the `Any` with a foreach though.